ui/src/components/dialog/__tests__/QDialog.cy.js

```diff
@@ -1,21 +1,131 @@
+import DialogWrapper from './DialogWrapper.vue'
+import { nextTick, ref } from 'vue'
+import { vModelAdapter } from '@quasar/quasar-app-extension-testing-e2e-cypress'
+
+function mountQDialogWrapper (options) {
+  return cy.mount(DialogWrapper, options)
+}
+
+function getHostElement () {
+  return cy.dataCy('dialog-card')
+}
+
+function closeDialogViaBackdrop () {
+  return cy.get('.q-dialog__backdrop').click({ force: true })
+}
+
+// TODO: this only works in `withinDialog` context,
+// as when using it outside it `.root()` will yeld "html" tag
+// and Cypress won't be able to find anything with `.closest('body')`
+// since it only searches upwards
+function closeDialogViaEscKey () {
+  // Official way to escape `within` context
+  // https://docs.cypress.io/api/commands/within#Temporarily-escape
+  return cy.root().closest('body').type('{esc}')
+}
+
+// TODO: we'll add this check to the AE helper to automatically test persistent dialogs
+// TODO: take into consideration that seamless dialogs have always 'aria-modal' false
+// TODO: take into consideration that dialogs without backdrop have always 'aria-modal' false
+// Check if there's a more reliable way to check for persistent dialogs
+function assertPersistentDialogExists () {
+  cy.dataCy('dialog-form').should('not.have.attr', 'aria-modal', 'false')
+}
+
 describe('Dialog API', () => {
   describe('Props', () => {
     describe('Category: behavior', () => {
       describe('(prop): persistent', () => {
-        it.skip(' ', () => {
-          //
+        it('should display a persistent dialog', () => {
+          const model = ref(true)
+          mountQDialogWrapper({
+            props: {
+              ...vModelAdapter(model),
+              persistent: true
+            }
+          })
+
+          cy.withinDialog({
+            persistent: true,
+            fn: () => {
+              closeDialogViaBackdrop()
+            }
+          })
+
+          assertPersistentDialogExists()
+
+          cy.withinDialog({
+            persistent: true,
+            fn: () => {
+              closeDialogViaEscKey()
+            }
+          })
+
+          assertPersistentDialogExists()
         })
       })
 
       describe('(prop): no-esc-dismiss', () => {
-        it.skip(' ', () => {
-          //
+        it('should not allow closing the dialog with the escape key', () => {
+          const model = ref(true)
+          mountQDialogWrapper({
+            props: {
+              ...vModelAdapter(model),
+              noEscDismiss: true
+            }
+          })
+
+          cy.withinDialog({
+            persistent: true,
+            fn: () => {
+              closeDialogViaEscKey()
+            }
+          })
+
+          assertPersistentDialogExists()
+
+          cy.wrap()
+            .then(async () => {
+              // Care as you need to wait setProps to complete its execution
```

Comment:
`Cypress.vueWrapper.setProps()` is an async function
If you just call it, without marking the `then` callback as `async`, Cypress wouldn't await for it to complete before proceeding with other commands, possibly breaking the test or causing a race condition

Check out if VTU or Cypress already provide an helper which minimize this category of errors
If not, you can try to create your own which automate the `cy.wrap().then(async () => {})` bit, even if the user would then still need to manually await its promises

If it's generic enough, we can add it to Cypress AE later on

Reply:
What do you think about this?
```ts
function getPreviousElement (fn) {
  return cy.wrap()
    .then(async () => {
      await fn()
    })
}
```
I wrote something like this but did not find any real benefit to using it instead of just calling `cy.wrap`, so I continued to use `cy.wrap`. What do you think?

Reply:
The name feels strange, why using "previous" or "previous element"?
`cy.wrap` doesn't actually touch in any way anything "previous" or an element

Reply:
Agreed. `cy.getPrevious` is not a good name here. Perhaps something like `getFulfilled`, `wrapAndFulFill`, `wrapAndThen` or something better in this case should be a better name. However, like I said in my previous comment, I don't think it serves some real benefit adding it here. However, I think it could be good in the AE. What do you think?

Reply:
If you didn't find it useful and we're not using it at least in our own tests, then let's avoid adding it in the AE for now

Let's try to add only what we know is actually useful